src/pkg/api/auth/auth.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2024 Defense Unicorns
+// SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+package auth
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/defenseunicorns/uds-runtime/src/pkg/api/auth/incluster"
+	"github.com/defenseunicorns/uds-runtime/src/pkg/api/auth/local"
+	"github.com/defenseunicorns/uds-runtime/src/pkg/config"
+)
+
+// Very limited special chars for git / basic auth
+// https://owasp.org/www-community/password-special-characters has complete list of safe chars.
+const randomStringChars = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ~-"
+
+// UserResponse is the response for the /auth endpoint
+type UserResponse struct {
+	Group             string `json:"group"`
+	PreferredUsername string `json:"preferred-username"`
+	Name              string `json:"name"`
+	InClusterAuth     bool   `json:"in-cluster-auth"`
+}
+
+// Configure sets the config vars for local or in-cluster auth
+func Configure() {
+	// check for local auth first
+	localAuthEnabled, err := strconv.ParseBool(strings.ToLower(os.Getenv("LOCAL_AUTH_ENABLED")))
+	if err != nil {
+		slog.Warn("invalid value for LocalAuthEnabled, must be 'true' or 'false'. Defaulting to 'true'")
+		localAuthEnabled = true
+	}
+
+	config.LocalAuthEnabled = localAuthEnabled
+	if localAuthEnabled {
+		slog.Info("Local auth enabled")
+		token, err := randomString(96)
+		if err != nil {
+			slog.Error("Failed to generate local auth token")
+			os.Exit(1)
+		}
+		local.AuthToken = token
+		return
+	}
+
+	// If local auth is disabled, check for in-cluster auth
+	inClusterAuthEnabled, err := strconv.ParseBool(strings.ToLower(os.Getenv("IN_CLUSTER_AUTH_ENABLED")))
+	if err != nil {
+		slog.Warn("invalid value for InClusterAuthEnabled, must be 'true' or 'false'. Defaulting to 'false'")
+		inClusterAuthEnabled = false
+	}
+
+	if inClusterAuthEnabled {
+		config.InClusterAuthEnabled = inClusterAuthEnabled
+		slog.Info("In-cluster auth enabled")
+	}
+}
+
+// RequestHandler is the main handler for the /auth endpoint; it returns a userResponse struct
+// indicating whether the request was authenticated via local or in-cluster auth, and relevant user data
+func RequestHandler(w http.ResponseWriter, r *http.Request) {
+	resp := UserResponse{
+		InClusterAuth: false,
+	}
+	if config.LocalAuthEnabled && !local.Auth(w, r) {
+		// auth failed, response is already written
+		return
+	} else if config.InClusterAuthEnabled {
```

Comment:
nit/suggestion: the conditional on line 73 doesn't flow into the next statement on 76, so we should break them apart. If line 73 is truthy then we always return on 75, so there's no need for the `else`. 

```suggestion
	if config.LocalAuthEnabled && !local.Auth(w, r) {
		// auth failed, response is already written
		return
	}
	
	if config.InClusterAuthEnabled {
```

There's no hard rule here, but it's mentioned in Effective Go and Google's Go style guide. 

- https://go.dev/doc/effective_go#control-structures
- https://google.github.io/styleguide/go/decisions#indent-error-flow